src/git-unix/misc.ml

```diff
@@ -32,26 +32,6 @@ let is_dir path =
 let open_pool = Lwt_pool.create 200 (fun () -> Lwt.return ())
 let mkdir_pool = Lwt_pool.create 1 (fun () -> Lwt.return ())
 
-(* XXX(samoht): Files smaller than this are loaded using [read].
-
-   Use of mmap is necessary to handle PACK files efficiently. Since
-   these are stored in a weak map, we don't run out of open files if
-   we keep accessing the same one.
-
-   Using read is necessary to handle references, since these are
-   mutable and can't be cached. Using mmap here leads to hitting the
-   OS limit on the number of open files.
-
-   This threshold must be larger than the size of a reference.
-
-   XXX(dinosaure): About the PACK file, we use directly mmap by the
-   [MAPPER] abstraction to compute efficiently all. I don't know if
-   it's reliable to keep this behaviour when we ensure than we use
-   [read] only for the loose object and mmap for the PACK file.
```

Comment:
@dinosaure do you know if this comment is still up-to-date? What happen if one tries to open the same file a lot? Do we run out of inodes? Is seems that we don't have that weak cache of fd anymore. See https://github.com/mirage/ocaml-git/pull/143 for the initial PR.

Reply:
To the current implementation the PACK was handle in 2 two:
1) when we meet a __new__ PACK file (without the associated IDX file), we read entirely the PACK file (with `read` syscall) continuously to produce the IDX file and calculate some stuff.
2) for all others, we use only the [MAPPER] abstraction (and, by this way, the `mmap` syscall) and keep file-descriptors of these PACK files along than the program run (so, one file-descriptor per PACK file - exactly one for the IDX file and one for the PACK file)

This is only about PACK file. And it's why, then, I explained it could be relevant to keep this for loose (big) object like blob. So we can delete the pool of `open` if you want when, only for PACK files, we use directly `mmap` - see [Unpack.Decoder](https://github.com/mirage/ocaml-git/blob/master/src/git/unpack.ml#L1394) and specifically [optimized_get](https://github.com/mirage/ocaml-git/blob/master/src/git/unpack.ml#L1964).

Reply:
So, I believed this comments stills up-to-date :) .

Reply:
But that `mmap_threashold` variable is only used when doing a test-and-set, e.g. for modifying references. I don't understand how this is used for large blobs.

Reply:
Ah, yeah. You are true, so if you delete `test_and_set`, you should delete `mmap_ threshold` because ... And because the reference size should not upper than `mmap_threshold`, it's useless to keep it.